services/device-registry-base/src/main/java/org/eclipse/hono/deviceregistry/service/device/AbstractRegistrationService.java

```diff
@@ -227,7 +405,41 @@ public final void setTenantInformationService(final TenantInformationService ten
                                 final RegistrationResult deviceResult = deviceInfoTracker.result();
                                 final RegistrationResult gatewayResult = gatewayInfoTracker.result();
 
-                                if (!isDeviceEnabled(deviceResult)) {
+                                if (deviceResult.isNotFound() && !gatewayResult.isNotFound()
+                                        && isDeviceEnabled(gatewayResult) && hasAuthorityForAutoRegistration(gatewayResult)) {
+
+                                    final Device device = new Device()
+                                        .setEnabled(true)
+                                        .setAutoProvisioned(true)
+                                        .setVia(Collections.singletonList(gatewayId));
+
+                                    final JsonArray memberOf = gatewayResult.getPayload().getJsonObject(RegistrationConstants.FIELD_DATA)
+                                            .getJsonArray(RegistryManagementConstants.FIELD_MEMBER_OF);
+                                    if (memberOf != null && !memberOf.isEmpty()) {
+                                        device.setViaGroups(memberOf.getList());
+                                    }
+
+                                    LOG.debug("auto-provisioning device {} for gateway {}", deviceId, gatewayId);
+                                    final Span childSpan = TracingHelper.buildChildSpan(tracer, span.context(), "auto-provisioning", Constants.PROTOCOL_ADAPTER_TYPE_DEVICE_REGISTRY)
```

Comment:
This span isn't getting finished below.

Reply:
I think this should be fixed now by the most recent refactoring.